esp/src/src-react/components/forms/CheckPermissions.tsx

```diff
@@ -0,0 +1,145 @@
+                const { Users, Groups } = FilePermissionResponse;
+                const sortedUsers = Users?.User?.sort((a, b) => a.username.localeCompare(b.username));
+                const sortedGroups = Groups?.Group?.sort((a, b) => a.name.localeCompare(b.name));
+
+                setUserOptions(sortedUsers?.map((user: { username: any; }) => {
+                    return { key: user.username, text: user.username };
+                }) ?? []);
+
+                setGroupOptions(sortedGroups?.map(group => {
+                    return { key: group.name, text: group.name };
+                }) ?? []);
+            } catch (error) {
+                logger.error(error);
+            }
+        };
+
+        fetchData();
+    }, []);
+
+    const closeForm = React.useCallback(() => {
+        reset(defaultValues);
+        setShowForm(false);
+        setFilePermissionResponse("");
+    }, [reset, setShowForm]);
+
+    const onSubmit = async (data: CheckPermissionsFormValues) => {
+        try {
+            const { FileName, SelectedValue } = data;
+            logger.info(`Checking permissions for file ${FileName} for selected value: ${SelectedValue}`);
+            const response = await FilePermission({
+                request: {
+                    FileName,
+                    UserName: SelectedValue,
+                    GroupName: SelectedValue
+                }
+            });
+            setFilePermissionResponse(response.FilePermissionResponse.UserPermission);
+        } catch (error) {
+            logger.error(error);
+            setErrorMessage("Error occurred while fetching file permissions.");
+            setShowError(true);
+        }
+    };
+
+    return (
+        <MessageBox show={showForm} setShow={closeForm} title={nlsHPCC.CheckFilePermissions} minWidth={400}
+            footer={<>
+                <PrimaryButton text={nlsHPCC.Submit} onClick={handleSubmit(onSubmit)} />
+                <DefaultButton text={nlsHPCC.Cancel} onClick={closeForm} />
+            </>}>
+            <Controller
+                control={control} name="FileName"
+                render={({
+                    field: { onChange, name: fieldName, value },
+                    fieldState: { error }
+                }) => <TextField
+                        name={fieldName}
+                        onChange={onChange}
+                        label={nlsHPCC.Name}
+                        value={value}
+                        errorMessage={error && error?.message}
+                    />}
+                rules={{
+                    required: nlsHPCC.ValidationErrorRequired
+                }}
+            />
+            <Dropdown
+                label={nlsHPCC.Users}
+                options={userOptions}
+                selectedKey={watch("SelectedValue")}
+                onChange={(ev, option) => {
+                    setValue("SelectedValue", option?.key.toString() || "");
```

Comment:
Similarly, the dojo version will blank out the other field when one changes, ie selecting a User will unset the selected Group if one was selected.